src/traits.rs

```diff
@@ -128,3 +128,110 @@ where
     T: Debug + Clone + HasAabb + SupportFunction<Point = <<Self as HasAabb>::Aabb as Aabb>::Point>,
 {
 }
+
+/// Trait used for interpolation of values
+///
+/// ## Type parameters:
+///
+/// - `S`: The scalar type used for amount
+pub trait Interpolate<S> {
+    /// Interpolate between `self` and `other`, using amount to calculate how much of other to use.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `amount`: amount in the range 0. .. 1.
+    /// - `other`: the other value to interpolate with
+    ///
+    /// ## Returns
+    ///
+    /// A new value approximately equal to `self * (1. - amount) + other * amount`.
+    fn interpolate(&self, other: &Self, amount: S) -> Self;
+}
+
+/// Trait used for interpolation of translation only in transforms
+pub trait TranslationInterpolate<S> {
+    /// Interpolate between `self` and `other`, using amount to calculate how much of other to use.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `amount`: amount in the range 0. .. 1.
+    /// - `other`: the other value to interpolate with
+    ///
+    /// ## Returns
+    ///
+    /// A new value approximately equal to `self * (1. - amount) + other * amount`.
+    fn translation_interpolate(&self, other: &Self, amount: S) -> Self;
+}
+
+mod interpolate {
+    use super::{Interpolate, TranslationInterpolate};
+    use cgmath::{BaseFloat, Basis2, Basis3, Decomposed, Quaternion, Rad, Vector2};
+    use cgmath::prelude::*;
+
+    impl<S> Interpolate<S> for Quaternion<S>
+    where
+        S: BaseFloat,
+    {
+        fn interpolate(&self, other: &Self, amount: S) -> Self {
+            self.lerp(*other, amount)
+        }
+    }
+
+    impl<S> Interpolate<S> for Basis3<S>
+    where
+        S: BaseFloat,
+    {
+        fn interpolate(&self, other: &Self, amount: S) -> Self {
+            Basis3::from(
+                Quaternion::from(*self.as_ref()).lerp(Quaternion::from(*other.as_ref()), amount),
```

Comment:
this doesn't take into account translation/scale?

Reply:
Basis3 only express a rotation from what I've seen?

Reply:
According to the cgmath docs: `A three-dimensional rotation matrix.`

Reply:
I've gone through cgmath code now, and there is no way to make Basis3 any other than a rotation matrix, so there is only rotation in there to interpolate.